src/main/java/nextstep/subway/path/domain/SubwayFare.java

```diff
@@ -0,0 +1,79 @@
+package nextstep.subway.path.domain;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+
+public class SubwayFare {
+    public static final int DEFAULT_FARE_VALUE = 1250;
+    public static final SubwayFare DEFAULT_FARE = new SubwayFare(DEFAULT_FARE_VALUE);
+
+    private static final int MAX_DISCOUNT_RATE = 100;
+    private static final int MIN_DISCOUNT_RATE = 0;
+    private static Map<Integer, SubwayFare> cacheMap = new HashMap<>();
+    private int value;
+
+    private SubwayFare() {
+        this.value = 0;
+    }
+
+    private SubwayFare(int value) {
+        if (value < 0) {
+            throw new IllegalArgumentException("운임은 음수가 될 수 없습니다.");
+        }
+        this.value = value;
+    }
+
+    public static SubwayFare of(Integer value) {
+        if (cacheMap.containsKey(value)) {
+            return cacheMap.get(value);
+        }
+        SubwayFare newSubwayFare = new SubwayFare(value);
+        cacheMap.put(value, newSubwayFare);
+        return newSubwayFare;
+    }
+
+    public SubwayFare plus(int plusValue) {
+        return new SubwayFare(value + plusValue);
```

Comment:
소소하지만 와 같이 불필요한 객체 생성을 방지 할 수 있어요.
```suggestion
          if(other == 0 ) {
              return this;
}
        return new SubwayFare(value + plusValue);
```

Reply:
👍